source/common/upstream/eds.cc

```diff
@@ -136,14 +146,16 @@ bool EdsClusterImpl::updateHostsPerLocality(const uint32_t priority, const HostV
   // out of the locality scheduler, we discover their new weights. We don't currently have a shared
   // object for locality weights that we can update here, we should add something like this to
   // improve performance and scalability of locality weight updates.
-  if (updateDynamicHostList(new_hosts, *current_hosts_copy, hosts_added, hosts_removed) ||
+  if (overprovisioning_factor.has_value() ||
+      updateDynamicHostList(new_hosts, *current_hosts_copy, hosts_added, hosts_removed) ||
```

Comment:
Hm, it looks like we call this function with the overprovisioning factor if it's present in the config.  Can we avoid doing an update of the old and new overprovisioning factors are the same?

Reply:
Done.

Reply:
we cannot? if the overprovisioning factor is changed, we have to update the host_set. 
on the contrary, if overprovisioning factor doesn't change, but hosts list changed, we have to keep the original behavior.